lib/msf/core/db_manager/service.rb

```diff
@@ -1,12 +1,16 @@
 module Msf::DBManager::Service
   # Deletes a port and associated vulns matching this port
-  def del_service(wspace, address, proto, port, comm='')
-
-    host = get_host(:workspace => wspace, :address => address)
-    return unless host
+  def delete_service(opts)
+    raise ArgumentError.new("The following options are required: :ids") if opts[:ids].nil?
 
   ::ActiveRecord::Base.connection_pool.with_connection {
-    host.services.where({:proto => proto, :port => port}).each { |s| s.destroy }
+    deleted = []
+    opts[:ids].each do |service_id|
+      service = Mdm::Service.find(service_id)
+      deleted << service.destroy
```

Comment:
I noticed this delete doesn't have the `rescue` block with the forced called to `delete`. Did you determine that was unnecessary?

Reply:
That's fine. We should create a backlog item to address the others that exist with the fall back to delete for consistency.

Reply:
I actually thought about it more, and I feel like the current `rescue` method makes the most sense. If we can't delete due to failing callbacks (most often caused by missing records that are linked to this one) then we will not be able to delete at all. The only method would be to manually delete it from the database.

This current method does log an error and gets it out of the DB. The error message could be a little bit more descriptive, however. We also should consider how we will tell the client this happened when running remotely.